### ISSUE TITLE
bug: Videos buffer unless the data connection is changed or RVX is closed and reopened

### ISSUE THREAD

**Issue by ArcherLegend** [anddea/revanced-patches#848]
### Type

Error at runtime

### Tools used

rvx-builder

### Application

Youtube 19.16.39
Patches versions ranging from 2.226.0 to 2.229.0 (latest release to date)

### Bug description

After RVX is started for the first time when turning on the phone (or after not being used for a few hours) if you start playing a video it buffers without loading. The workarounds are the following 2:
1. close RVX and open it again
2. disconnect from wifi/data and reconnect

Once that is done every video starts playing normally till the device is rebooted or RVX is not used for a few hours.

Spoof player parameters is turned on and the chosen client (iOS, Android TV, Android VR) does not fix the issue (my settings are attached in the additional context section).
The bug appears whether MicroGMS is restricted or unrestricted.

After cleanly installing both MicroGMS and RVX the issue may not appear instantly but it may take 1 or 2 days before occuring and once that happens for the first time it keeps doing what I described.

### Additional context

"append_time_stamp_information": false,
"change_start_page": "open.subscriptions",
"default_video_quality_mobile": 1080,
"default_video_quality_wifi": 1080,
"disable_ambient_mode": true,
"disable_ambient_mode_in_fullscreen": true,
"disable_hdr_video": true,
"disable_speed_overlay": true,
"disable_splash_animation": true,
"enable_gradient_loading_screen": true,
"enable_seekbar_tapping": false,
"enable_swipe_brightness": false,
"enable_swipe_lowest_value_auto_brightness": false,
"enable_swipe_press_to_engage": true,
"enable_watch_panel_gestures": true,
"gms_show_dialog": false,
"hide_carousel_shelf": true,
"hide_clip_button": true,
"hide_comment_create_shorts_button": true,
"hide_comment_thanks_button": true,
"hide_comment_timestamp_and_emoji_buttons": true,
"hide_comments_by_members": true,
"hide_download_button": true,
"hide_feed_survey": false,
"hide_handle": false,
"hide_latest_posts": false,
"hide_latest_videos_button": false,
"hide_mix_playlists": true,
"hide_movie_shelf": true,
"hide_navigation_notifications_button": true,
"hide_navigation_shorts_button": true,
"hide_notify_me_button": true,
"hide_player_autoplay_button": false,
"hide_player_youtube_music_button": true,
"hide_quick_actions_more_button": true,
"hide_quick_actions_open_mix_playlist_button": true,
"hide_quick_actions_open_playlist_button": true,
"hide_remix_button": true,
"hide_rewards_button": true,
"hide_shop_button": true,
"hide_shorts_shelf_history": true,
"hide_snack_bar": true,
"hide_thanks_button": true,
"hide_toolbar_create_button": true,
"hide_voice_search_button": true,
"remember_playback_speed_last_selected": false,
"remove_viewer_discretion_dialog": true,
"spoof_streaming_data": true,
"spoof_streaming_data_type": "ios",
"swipe_overlay_background_alpha": 0,
"swipe_overlay_rect_size": 50,
"switch_create_with_notifications_button": false,
"ryd_user_id": "act0lY7slv00JLqJpNEYa8TvpBBAhoN6fIKr"

### Device Environment

Galaxy S10+ (Exynos 9820 variant) Android 10 One UI 2.1 build version QP1A.190711.020.G975FXXU8CTG4

### Acknowledgements

- [X] This issue does not reproduce on unpatched YouTube or YT Music.
- [X] This issue is not a duplicate of an existing bug report.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.
- [X] I have written the title and contents in English.

**Comment by Francesco146** [comment]
we share same code base as RVX, please open the issue there and make sure you're using the dev release, not the stable one, since it has the latest fixes

**Comment by ArcherLegend** [comment]
@Francesco146 the issue happens both on the release and the dev releases.

**Comment by Francesco146** [comment]
try taking a look at the RVX issue center if it's known. since it's not related to a unique feature of this repo, it's best to open it there

**Comment by ArcherLegend** [comment]
@anddea @Francesco146 I opened this same issue [here](https://github.com/inotia00/ReVanced_Extended/issues/2416#issue-2560089972) and it being monitored by inotia00, as you can read he said that he had made some changes to the spoof streaming code for android VR and TV in the latest dev branch and he is waiting for my confirmation to know if the issue is fixed before adding those changed to the iOS client. If the issue is fixed and the changes are made to the iOS client in his next dev build I wish that your fork of RVX dev builds gets said changes as well.
Let me know.

**Comment by Francesco146** [comment]
do what he says (?) your issue is only yours, we can't reproduce on our end

**Comment by ArcherLegend** [comment]
> do what he says (?) your issue is only yours, we can't reproduce on our end

https://github.com/inotia00/ReVanced_Extended/issues/2416#issuecomment-2387993331
